nestjs-migration/src/post/post.controller.spec.ts

```diff
@@ -0,0 +1,215 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { PostController } from "./post.controller";
+import { PostService } from "./post.service";
+import {  CreatePostDto } from "./dto/create-post.dto";
+import { ServerError } from "../common/exceptions/server-error.exception";
+import { POST_ERROR_MESSAGES } from "./constant/post.constants";
+
+describe("PostController", () => {
+	let postController: PostController;
+	let postService: PostService;
+
+	const mockPostService = {
+		createPost: jest.fn(),
+		findPostHeaders: jest.fn(),
+		findPostTotal: jest.fn(),
+		findPost: jest.fn(),
+		updatePost: jest.fn(),
+		deletePost: jest.fn(),
+	};
+
+	let mockReq: any;
+	let mockUserId: number;
+	let mockPostId: number;
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			controllers: [PostController],
+			providers: [
+				{
+					provide: PostService,
+					useValue: mockPostService,
+				},
+			],
+		}).compile();
+
+		postController = module.get<PostController>(PostController);
+		postService = module.get<PostService>(PostService);
+
+		mockUserId = 1;
+		mockPostId = 9;
+		mockReq = {
+			user: { userId: mockUserId },
+		} as any;
+	});
+
+	describe("POST /post", () => {
+		const mockCreatePostDto: CreatePostDto = {
+			title: "Test Title",
+			content: "Test Content",
+      authorId: mockUserId,
+			doFilter: true,
+		};
+
+		it("게시물 생성 성공", async () => {
+			jest.spyOn(mockPostService, "createPost").mockResolvedValue(
+				mockPostId
+			);
+			const result = await postController.handlePostCreate(
+				mockCreatePostDto,
+				mockReq
+			);
+
+			expect(postService.createPost).toHaveBeenCalledWith(
+				{...mockCreatePostDto, author_id: mockUserId}
```

Comment:
이렇게 수정하니 test 원할하게 통과됩니다.
```suggestion
				{...mockCreatePostDto, authorId: mockUserId}
```